### PR TITLE
Hubspot

### PR DESCRIPTION
**Pull request by jucallej**
This is using the work of @drwlrsn https://github.com/calibreapp/react-live-chat-loader/pull/130 but I moved it into my own fork, as I didn't have permissions to push to your repo @drwlrsn hope that's ok. Thanks for the work 🙏

## What does this PR introduce?
In a few bullet points, please describe the changes this Pull Request makes. e.g.:

- Adds hubspot support

## Related issues
Are there any related [issues or feature requests](https://github.com/calibreapp/react-live-chat-loader/issues) this work will resolve? Please mention them here. e.g.:

- closes #https://github.com/calibreapp/react-live-chat-loader/pull/130

## Screenshots
167025070-2d1eb53b-863e-4d04-90e3-3f7d3e9341f2

### CODE REVIEW

**Comment by jucallej** [comment]
> [CLA assistant check](https://cla-assistant.io/calibreapp/react-live-chat-loader?pullRequest=281) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/calibreapp/react-live-chat-loader?pullRequest=281) before we can accept your contribution.**1** out of **2** committers have signed the CLA.✅ jucallej❌ drwlrsnYou have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/calibreapp/react-live-chat-loader?pullRequest=281) it.

@drwlrsn when you have some time, would you ming signing that CLA (it just asks clicking a checkbox) to be able to merge this? Thank you 🙏

**Comment by jucallej** [comment]
Having a look now to update this with an example and updating the readme as per the lastest changes on the repo

**Comment by jucallej** [comment]
> Having a look now to update this with an example and updating the readme as per the lastest changes on the repo

Just added the example and the readme instructions. This is how it looks locally:
Screenshot 2023-06-14 at 11 57 49
Screenshot 2023-06-14 at 11 57 59

**Comment by jucallej** [comment]
Hey @drwlrsn 👋, when you have some time (if at all possible), would you be ok to sign https://github.com/calibreapp/react-live-chat-loader/pull/281#issuecomment-1590928909🙏

@benschwarz and @thefoxis when you have some time, do you have any feedback for this PR? 😀

**Comment by benschwarz** [comment]
@jucallej Thank you for this work 🙌

@robmorieson will complete a code review. 

We'll have to figure something out for the Hubspot demo key, at the moment I'm not sure how to handle that!

**Comment by benschwarz** [comment]
@drwlrsn Hello! @jucallej completed work on the react-live-chat-loader Hubspot branch that you commenced a while back. Could you sign the CLA so we can merge the completed work? Thank you 🤗

**Review by robmorieson — CHANGES_REQUESTED** [review]
@jucallej Thanks for submitting this PR 🙌  The code itself looks solid, I've just left a few review comments/suggestions.  

In regards to the implementation itself, I did run into some runtime errors while testing on mobile (iOS 16.4 simulator) so that might require some investigating: 

Simulator Screenshot - iPhone 14 Pro Max - 2023-06-27 at 11 12 53

File: src/providers/hubSpot.ts
Comment (jucallej) [reply src/providers/hubSpot.ts]: Great catch! I've just updated it in the last commit 😀

File: src/providers/hubSpot.ts
Comment (jucallej) [reply src/providers/hubSpot.ts]: Great catch! I've just updated it in the last commit 😀

File: src/components/HubSpot/index.tsx
Comment (jucallej) [reply src/components/HubSpot/index.tsx]: Sounds good! I've just changed that in the last commit 😀

File: src/components/HubSpot/index.tsx
Comment (jucallej) [reply src/components/HubSpot/index.tsx]: Great catch! I've just updated it in the last commit 😀

**Comment by jucallej** [comment]
> @jucallej Thanks for submitting this PR 🙌 The code itself looks solid, I've just left a few review comments/suggestions.
> 
> In regards to the implementation itself, I did run into some runtime errors while testing on mobile (iOS 16.4 simulator) so that might require some investigating:
> 
> Simulator Screenshot - iPhone 14 Pro Max - 2023-06-27 at 11 12 53

Thanks for the review 🙌! I've refactored it to use a method called isHubspotWidgetDefined and use it throughout, and seems to work at least on safari on mac. Is it still failing with the latest changes? What emulator are you using to test? 😀

File: src/providers/index.ts
Comment (benschwarz) [inline src/providers/index.ts:7]:
```suggestion
export { default as hubSpot } from './hubSpot'
```

File: README.md
Comment (cvan) [inline README.md:381]:
```suggestion
```

This causes a regression that hides the **Adding a Provider** content in a collapsed `<details>` tag.

Comment (jucallej) [reply README.md]: Thanks for the catch! I've just committed your suggestion 🙌

**Review by robmorieson — APPROVED** [review]

**Comment by robmorieson** [comment]
@jucallej finally got around to testing (apologies for the _very long_ wait!) — all looks good - thanks!  However, unfortunately it appears we're still waiting for to @drwlrsn sign the CLA before we can merge this one...

**Comment by robmorieson** [comment]
Merging this one as it's been several months since the callouts for CLA approval. 

@drwlrsn — if any objections, please let us know.

Otherwise, thank you @drwlrsn and @jucallej for your contribution here! 🙌

**Pull request merged**
